pkg/providers/tinkerbell/hardware/translate.go

```diff
@@ -48,6 +48,9 @@ func Translate(reader MachineReader, writer MachineWriter, validator MachineVali
 		return fmt.Errorf("read: invalid hardware: %v", err)
 	}
 
+	for _, modify := range modifiers {
+		machine = modify(machine)
+	}
```

Comment:
What do you think about decorating the `MachineWriter` instead of passing modifiers?